File: fullstack-fintech-app/blacklist-service/internal/domain/model/black_listed_wallet.go
package model

import ("time")

type BannedReasons string

const (
	SuspiciousActivity         BannedReasons = "SUSPICIOUS_ACTIVITY"
	TermsOfServiceViolation    BannedReasons = "TERMS_OF_SERVICE_VIOLATION"
	FraudulentActivity         BannedReasons = "FRAUDULENT_ACTIVITY"
	HarassmentOrBullying       BannedReasons = "HARASSMENT_OR_BULLYING"
	InappropriateContent       BannedReasons = "INAPPROPRIATE_CONTENT"
	PlatformManipulation       BannedReasons = "PLATFORM_MANIPULATION"
	IdentityVerificationFailure BannedReasons = "IDENTITY_VERIFICATION_FAILURE"
	SpammingOrSolicitation     BannedReasons = "SPAMMING_OR_SOLICITATION"
	MultipleUserReports        BannedReasons = "MULTIPLE_USER_REPORTS"
	ModeratorAction            BannedReasons = "MODERATOR_ACTION"
)

type BlackListedWallet struct {
	ID              uint           `gorm:"primaryKey;autoIncrement" json:"id"`
	WalletID        uint           `gorm:"not null" json:"wallet_id"`
	BankBannedReason BannedReasons `gorm:"type:varchar(200);not null" json:"bank_banned_reason"`
	IsBlock         bool           `gorm:"not null" json:"is_block"`
	Timestamp       time.Time      `gorm:"type:timestamp;not null" json:"timestamp"`
	CreatedAt       time.Time      `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt       time.Time      `gorm:"autoUpdateTime" json:"updated_at"`
}

File: fullstack-fintech-app/resiliences-service/Models/BlackListedWallet.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using resiliences_service.Enums;

namespace resiliences_service.Models
{
    public class BlackListedWallet
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public uint Id { get; set; }

        [Required]
        public uint WalletId { get; set; }

        [Required]
        [Column(TypeName = "varchar(200)")]
        public BannedReasons BankBannedReason { get; set; }

        [Required]
        public bool IsBlock { get; set; }

        [Required]
        public DateTime Timestamp { get; set; }

        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [DatabaseGenerated(DatabaseGeneratedOption.Computed)]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}